README.md

```diff
@@ -1,72 +1,304 @@
-
-For instance:
-
-`otel-collector.yaml`:
-```yaml
-receivers:
-  opencensus:
-
-exporters:
-  logging:
-    verbosity: detailed
-
-  azuremonitor:
-    maxbatchsize: 100
-    maxbatchinterval: 10s
-    # REPLACE WITH YOUR KEY:
-    instrumentation_key: YOUR-KEY-HERE
-
-service:
-  telemetry:
-    logs:
-      level: "warn"
-  pipelines:
-    traces:
-      receivers: [opencensus]
-      exporters: [azuremonitor]
+
+OpenTelemetry building blocks to instrument [Lura](https://github.com/luraproject/lura) / [KrakenD](https://krakend.io) 
+based API Gateways.
+
+[Apache Licnese 2.0](.LICENSE)
+
+## Example
+
+For a quick look at the observability the library can provide, check the 
+[example documentation](./example/README.md).
+
+## Configuration from Lura's [ServiceConfig](https://github.com/luraproject/lura/blob/master/config/config.go)
+
+In order to configure the open telemetry stack to instrument the API Gateway, a 
+new entry must be added to the `ExtraConfig` root propertry of `ServiceConfig` 
+using the `telemetry/opentelemetry` key, with the `krakend-otel`'s configuration.
+
+## `krakend-otel` Configuration
+
+In the configuration we find the following root entries:
+
+- `service_name`: to provide a custom name for this service. However if there is 
+  a `ServiceConfig.Name` already set, that will be the one used.
+- `exporters`: in this section we map our own "custom names" for different 
+  exporters configurations. This allows us to use the same `kind` of exporter,
+  with different port / host configurations, and report the same metric / traces
+  to different systems (imagine migrating from a self hosted grafana stack to the
+  cloud version, and that for a while you want to report to both places before
+  making the switch), of report to different `kind` of exporters (migrating from one cloud 
+  observability system to a different one).
+- `layers`: in this section we can fine tune the amount of metrics / traces that 
+  we want to report at each "stage" of the processing pipeline: `router`, `pipe` or `backend`.
+- `instance`: in this section we select which exporters we want to use for metrics, and
+  traces, and the params to tweak the telemetry instance that will be used.
+- `skip_paths`: an option to define endpoints that we do not want to instrument. By default,
+    if it is not provided, it will skip the following "internal" paths:
+		- `/healthz`
+		- `/_ah/health`
+		- `/__debug`
+		- `/__echo`
+    If we want to instrument those paths too, just provide an empty string as endpoint to
+    skip.
+- `extra`: we leave this entry for any other "custom configuration" that a user of this 
+  library might want to add.
+
+In a visual way, this is the realation between the `exporters` configuration, and 
+how we select as `metric_providers` or `trace_providers`: 
+
+![krakend_otel_exporters.svg](./doc/krakend_otel_exporters.svg)
+
+### Exporters 
+
+Example:
+
+```json
+"exporters": {
+    "local_prometheus": {
```

Comment:
Thinking in external tools and config validation, I would rather avoid having a "free text" key, and convert exporters to an array instead. Something like:
```
"exporters": [
{
  "name": "my_prometheus_fashion_Exporter",
  "kind": "prometheus",
  "config": {
     "port": 9090,
     "process_metrics": true,
     "go_metrics": true
   }
 },
....
]```